### PR TITLE
[READY] pull out filter_params and add archived/unarchived filter to reports

### PR DESCRIPTION
**Pull request by jurre**
This should make it really easy to add more filters ^^

### CODE REVIEW

**Comment by jurre** [comment]
I've separated the two filter forms again, I think that's cleaner than doing a bunch of `if @filters.options[:archived]` or something like that

File: app/models/entry_filter.rb [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
+
+  def archived_options
+    [
+      [I18n.t("entry_filters.not_archived"), false],
+      [I18n.t("entry_filters.archived"), true]
     ]
   end
+end
+
+class EntryFilter
```
Comment (Marthyn) [inline app/models/entry_filter.rb]: Are you defining the class EntryFilter twice now?

File: config/locales/en.yml [hunk 2]
```diff
@@ -230,3 +220,17 @@ en:
...
+    billed_label: Billed or not
+    billed_empty: Billed or not
+    clients: Clients
+    from_date_label: From date
+    not_billed: Not billed
+    to_date_label: To date
+    projects: Projects
+    archived_empty: Archived and Unarchived
+    archived: Archived
+    not_archived: Not Archived
```
Comment (Marthyn) [inline config/locales/en.yml]: Could you put the entry_filters around the E in the locale? It's best to keep it as much in alphabetical order as we can.

File: app/models/entry_filter.rb (hunk 1 above)
Comment (jurre) [inline app/models/entry_filter.rb]: No, just reopening it

**Pull request merged**